src/main/java/seedu/financialplanner/cashflow/Budget.java

```diff
@@ -50,17 +54,22 @@ public static void deduct(double amount) {
         Budget.currentBudget -= amount;
     }
 
-    public static void load(double initial, double current) {
+    public static void load(double initial, double current, LocalDate savedDate) {
         initialBudget = initial;
         currentBudget = current;
+        // Resets budget if it is a new month or new year.
+        if (!currentDate.getMonth().equals(savedDate.getMonth()) || currentDate.getYear() != savedDate.getYear()) {
```

Comment:
possible to add exceptions if the currentDate is before savedDate

Reply:
added to loadbudget in loaddata